File: CodingSamples/DotNet/Fundamentals/Inheritance/ClassTypeTest/DemoApp1/Payroll/SalesPerson.cs
namespace Payroll
{
    //Defining SalesPerson as a derived class of Employee (base class)
    class SalesPerson : Employee
    {
        //automatic property
        public double Sales { get; set; }

        public SalesPerson(int h, float r, double s) : base(h, r)
        {
            Sales = s;
        }

        //method overriding - defining a method in a derived class
        //with override keyword whose declaration matches with a
        //virtual method defined in the base class
        public override double Income()
        {
            double amount = base.Income();
            if(Sales >= 20000)
                amount += 0.05 * Sales;
            return amount;
        }
    }
}

File: CodingSamples/Java/Fundamentals/Inheritance/SubclassTypeTest/demo1/payroll/SalesPerson.java
package payroll;

//define SalesPerson as a subclass of Employee (super-class)
public class SalesPerson extends Employee {
    
    private double sales;

    public SalesPerson(int h, float r, double s) {
        super(h, r);
        sales = s;
    }

    public double getSales() {
        return sales;
    }

    public void setSales(double value) {
        sales = value;
    }

    //method overriding - defining method in subclass whose declaration
    //matches with the declaration of same method in super-class
    public double income() {
        double amount = super.income();
        if(sales >= 25000)
            amount += 0.05 * sales;
        return amount;
    }
}
